cloud/terraform/azure_config_builder.py

```diff
@@ -46,12 +48,35 @@ def build_resources(self):
             self.__new_azure_virtual_network(instance)
             self.__new_azure_subnet(instance)
 
+            if 'vhd_uri' in instance:
+                self.__new_azure_image(instance)
+
             self.__new_azure_public_ip(instance)
             self.__new_azure_nic(instance)
             self.__new_azure_vm(instance)
 
         return self.resources_tf
 
+    def __new_azure_image(self, instance):
+        name = self.create_resource_name([instance['location'], 'image'])
+
+        instance['image_from_vhd'] = name
+
+        os_disk = {
+            'os_type': 'Linux',
+            'os_state': 'Generalized',
+            'blob_uri': instance['vhd_uri']
+        }
+
+        new_subnet = {
+            'name': name,
+            'location': instance['location'],
+            'resource_group_name': self.resource_group,
+            'os_disk': os_disk,
+        }
+
+        self.resources_tf['resource']['azurerm_image'][name] = new_subnet
```

Comment:
I think variable names are a bit confusing. You are assigning `new_subnet` to an `azurerm_image`. Is this correct?

Reply:
LOL - copy&paste effect! I will change it. Well spotted!